modules/openapi-generator/src/test/java/org/openapitools/codegen/go/GoClientCodegenTest.java

```diff
@@ -109,4 +119,26 @@ public void testFilenames() throws Exception {
         Assert.assertEquals(codegen.toApiFilename("Animal Farm Test"), "api_animal_farm_test_");
     }
 
+    @Test
+    public void testPrimitiveTypeInOneOf() throws IOException {
+        File output = Files.createTempDirectory("test").toFile();
+        output.deleteOnExit();
+
+        final CodegenConfigurator configurator = new CodegenConfigurator()
+                .setGeneratorName("go")
+                .setInputSpec("src/test/resources/3_0/oneOf_primitive.yaml")
+                .setOutputDir(output.getAbsolutePath().replace("\\", "/"));
+
+        DefaultGenerator generator = new DefaultGenerator();
+        List<File> files = generator.opts(configurator.toClientOptInput()).generate();
+        System.out.println(files);
+        files.forEach(File::deleteOnExit);
+
+        Path modelFile = Paths.get(output + "/model_example.go");
+        TestUtils.assertFileContains(modelFile, "Child1 *Child1");
+        TestUtils.assertFileContains(modelFile, "String *string");
+        TestUtils.assertFileContains(modelFile, "dst.String");
```

Comment:
Personally I would rather see `dst.Child2` here if you are using `$ref`.

If you want base types then we should prefer:
```
oneOf:
- $ref: '#/components/schemas/Child1'
- $ref: '#/components/schemas/Child2'
- type: integer
  format: int32
```
->
```
TestUtils.assertFileNotContains(modelFile, "dst.int");
```